### DISCUSSION TITLE
Output file formatting via "printf"-style format specifiers ("%d" etc)

### DISCUSSION THREAD

**Discussion by Pointy** [ImageMagick/ImageMagick#5249] [Help] [answered]
I've read multiple examples that suggest that printf-style formatting escapes (like "%d" or "%3d") should allow a command like `convert` working on a multiple-page PDF file to produce output files with names according to the requested formatting. However, when I try for example:

    convert -density 200 foo.pdf -quality 90 output-%3d.jpg

what I get are files named "output-%3d-0.jpg", "output-%3d-1.jpg", etc.

When I use "%d" and no width specifier, it works. I can't find anything in the documentation (so far at least) that gives a thorough summary of what formatting options are possible. I have definitely seen examples that use exactly the format I am using, and none of those samples mention that it doesn't work.

I'm on version 7.1.0-37 on MacOS, if that matters.

**Comment by fmw42** [comment, marked as answer]
You need zeros before 3d.

`convert -density 200 foo.pdf -quality 90 output-%03d.jpg`

should produce output-000.jpg, output-001.jpg, output-002.jpg ... etc

**Reply by Pointy** [reply]
Yep that does it, but I swear that there are many examples out there without the 0.  Thank you so much!
